sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SqlOperators.java

```diff
@@ -155,6 +156,25 @@ public static SqlFunction createZetaSqlFunction(String name, SqlTypeName returnT
         SqlFunctionCategory.USER_DEFINED_FUNCTION);
   }
 
+  /**
+   * Create a dummy SqlAggFunction of type OTHER_FUNCTION from given function name and return type.
+   * These functions will be executed by Beam CombineFns implemented in {@link
+   * org.apache.beam.sdk.extensions.sql.impl.transform.BeamBuiltinAggregations}.
+   */
+  public static SqlAggFunction createZetaSqlAggFunction(String name, RelDataType returnType) {
+    return new SqlAggFunction(
+        name,
+        null, // sqlIdentifier
+        SqlKind.OTHER_FUNCTION,
+        x -> returnType,
```

Comment:
I am not sure whether this is the right trigger to override the return type of an agg operator. Maybe the best to evaluate it is to support `AVG` in ZetaSQL. As I recall the return type is different for ZetaSQL and Calcite when input parameter type is `INT64` for `AVG`.